backend/functions/student/functions.js

```diff
@@ -56,8 +64,10 @@ const addStudentSurveyResponse = async (
       throw e;
     });
 
+  
   // Next, update each course record to add this student
-  const courseUpdates = courseCatalogNames.map((courseCatalogName, index) =>
+  const coursesToUpdate = courseCatalogNames.filter((crse) => crsesToAdd.includes(mapCatalogNameToCrseId(crse)))
```

Comment:
Try to avoid using `mapCatalogNameToCrseId` here, it's pretty expensive because it involves a network call for every catalog name